Formula/yq@3.rb

```diff
@@ -0,0 +1,30 @@
+class YqAT3 < Formula
+  desc "Process YAML documents from the CLI - Version 3"
+  homepage "https://github.com/mikefarah/yq"
+  url "https://github.com/mikefarah/yq/archive/3.4.1.tar.gz"
+  sha256 "73259f808d589d11ea7a18e4cd38a2e98b518a6c2c178d1ec57d9c5942277cb1"
+  license "MIT"
+
+  keg_only :versioned_formula
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/mikefarah/yq").install buildpath.children
+
+    cd "src/github.com/mikefarah/yq" do
+      system "go", "build", "-o", bin/"yq"
+      prefix.install_metafiles
+
+      (bash_completion/"yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=bash")
+      (zsh_completion/"_yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=zsh")
+      (fish_completion/"yq.fish").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=fish")
+    end
```

Comment:
If version 3.4.1 supports go modules, we can simplify it as we done it for yq 4:

```suggestion
    system "go", "build", "-ldflags", "-s -w", *std_go_args

    (bash_completion/"yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=bash")
    (zsh_completion/"_yq").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=zsh")
    (fish_completion/"yq.fish").write Utils.safe_popen_read("#{bin}/yq", "shell-completion", "-V=fish")
```

Reply:
The build works fine, but none of the completion commands work.

Although come to think of it - are they supposed to be included in a 'keg-only' version of yq?

Reply:
They are not afaik

Reply:
Ok I've simplified the install, I did have to specify `yq` as the go build output to build the `yq` binary.

All the brew tests worked (build from source, test, audit strict)